cairo/ethereum/cancun/state.cairo

```diff
@@ -1017,3 +1017,40 @@ func touch_account{poseidon_ptr: PoseidonBuiltin*, state: State}(address: Addres
     set_account(address, empty_account);
     return ();
 }
+
+func destroy_touched_empty_accounts{poseidon_ptr: PoseidonBuiltin*, state: State}(
+    touched_accounts: SetAddress
+) -> () {
+    alloc_locals;
+
+    // if current == end, return
+    let current = cast(touched_accounts.value.dict_ptr_start, DictAccess*);
+    let end = cast(touched_accounts.value.dict_ptr, DictAccess*);
+    if (current == end) {
+        return ();
+    }
+
+    let key = [current].key;
+    tempvar address = Address(key);
```

Comment:
if you didn't cast current into a DictAccess, key would already be of `address` type

Reply:
I dont actually think this cast is required

Reply:
done with [64df859](https://github.com/kkrt-labs/keth/pull/505/commits/64df859a75a641f23139e0488feef0ffd7153d75)